vm_supervisor/pool.py

```diff
@@ -125,6 +132,11 @@ async def stop(self):
             *(execution.stop() for vm_hash, execution in self.executions.items())
         )
 
+        # Stop instance snapshot executions in parallel:
+        await asyncio.gather(
+            *(self.snapshot_manager.stop_for(vm_hash) for vm_hash, execution in self.get_instance_executions())
+        )
```

Comment:
The SnapshotExecution instance should be stopped once the VM stops (is just that is doing now), and the snapshot manager should continue working because is the general manager for all the VMs. But yes, I agree, we can ensure to do a last snapshot when the VM is down to ensure to have the latest data.

Reply:
Okay. Could you encapsulate this gather operation in a method of `SnapshotManager`? I feel like the caller should just call `snapshot_manager.stop()` or something like that and be done with it.

Reply:
Done :heavy_check_mark: